app.rb

```diff
@@ -0,0 +1,160 @@
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person(role)
+    puts "Creating a new #{role}..."
+    puts 'Enter name:'
+    name = gets.chomp
+    puts 'Enter age:'
+    age = gets.chomp.to_i
+  
+    case role
+    when 'student'
+      if age < 18
+        puts 'Does the student have parent permission? (yes/no):'
+        permission_input = gets.chomp.downcase
+        parent_permission = permission_input == 'yes'
+      else
+        parent_permission = true
+      end
+  
+      person = Student.new(age, name, parent_permission: parent_permission)
+    when 'teacher'
+      puts 'Enter specialization:'
+      specialization = gets.chomp
+      person = Teacher.new(specialization, name: name, age: age)
+    end
+  
+    @people << person
+  end
+
+  def create_book
+    puts 'Creating a new book...'
+    puts 'Enter Author:'
+    title = gets.chomp
+    puts 'Enter Title:'
+    author = gets.chomp
+    book = Book.new(author, title)
+    @books << book
+  end
+
+  def create_rental
+    puts 'Creating a new rental...'
+    puts 'Enter person ID:'
+    person_id = gets.chomp
+    person = @people.find { |p| p.id == person_id }
+    if person.nil?
+      puts 'Person not found!'
+      return
+    end
+
+    puts 'Enter book title:'
+    book_title = gets.chomp
+    book = @books.find { |b| b.title == book_title }
+    if book.nil?
+      puts 'Book not found!'
+      return
+    end
+
+    puts 'Enter date (YYYY-MM-DD):'
+    date = gets.chomp
+
+    @rentals << Rental.new(date, person, book)
+  end
+
+  def list_all_rentals_for_person(person_id)
+    person = @people.find { |p| p.id == person_id.to_s }
+
+    if person.nil?
+      puts 'Person not found!'
+    else
+      puts "Rentals for #{person.name}:"
+      rentals = @rentals.select { |rental| rental.person.id == person.id }
+
+      if rentals.empty?
+        puts "No rentals found for #{person.name}."
+      else
+        rentals.each do |rental|
+          puts "Date: #{rental.date}, Book: #{rental.book.title}"
+        end
+      end
+    end
+  end
+
+  def main
+    loop do
+      puts 'Welcome to the Library App!'
+      puts 'Choose an option:'
+      puts '1. List all books'
+      puts '2. List all people'
+      puts '3. Create a person'
+      puts '4. Create a book'
+      puts '5. Create a rental'
+      puts '6. List all rentals for a given person ID'
+      puts '7. Quit'
+
+      choice = gets.chomp.to_i
+
+      case choice
+      when 1
+        list_all_books
+      when 2
+        list_all_people
+      when 3
+        role = role_from_user
+        create_person(role)
+      when 4
+        create_book
+      when 5
+        create_rental
+      when 6
+        puts 'Enter the person ID:'
+        person_id = gets.chomp.to_i
+        list_all_rentals_for_person(person_id)
+      when 7
+        break
+      else
+        puts 'Invalid option. Please try again.'
+      end
+    end
+  end
```

Comment:
As per project requirements, please make sure that the `main.rb` file serves as an entry point for the application. In your case, The `main` method should be moved to the `main.rb` file and should only be responsible for presenting the user with a list of options to perform.

![image](https://github.com/Ochiengsteven/school-library-oop/assets/38088818/8682e009-bcb1-4c23-8654-d7ec3d1d033d)


The `main.rb` should be similar to this but modified to suit your application.

```ruby
def main
  app = App.new()

  response = nil

  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

# 'main' method should be invoked at the end of the main.rb file.
main()
```

This will ensure that your application is well organized and every file has a single responsibility.